### PR TITLE
Fixes #2863: added delay and loader

### LINKED ISSUE #2863: Disable shipping methods when triggering shipping recalculation
The user experience in the checkout when shipping changes is a bit disconnected, as it's not always clear when shipping is being recalculated.  Whenever recalculation is triggered,  it'd be good be that the existing shipping methods, selected or not, should become disabled and existing selection reset while the user is informed of this progress.

### Expected behavior
To see that my cart is recalculating shipping rates and see have feedback on the process progress.

### Actual Behavior

Nothing happens until the rate is re-calculated and has been returned. This can be quick, but sometimes rates can take 5-10 second, which is enough time to get bored and click more stuff.

### Steps to Reproduce the Behavior
Add two shipping methods, switch back and forth. Observe, wait, observe...  Even if the rates are returning lightning quick, you'll start seeing the bad user experience quick enough. It's easy enough to mistake this for nothing have happened. 

### Versions
Node: 6.11.1
NPM: 5.3.0
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: folusho-fix-shipping-methods-not-configured-2265 (marketplace)
Docker: 17.06.1-ce

### PR DESCRIPTION
**Pull request by Akarshit**
Resolves #2863  
Impact: major
Type: bugfix

## Issue
The issue was basically a design issue in which the user wasn't given feedback of the updation of rates when another shipping method was selected.

## Solution
Added a loader which would be shown before the rates are calculated.
But in my testing, whenever I chose another method the calculation was fast enough that the loader wasn't visible in that time.
So I added a delay in the calculation of 200ms.

## Testing
1. Enable Shippo
2. But the `Example product` and go to the checkout screen.
1. Complete steps till filling of address
3. Click on various shipping methods and see the loader over the Review portion.

### CODE REVIEW

**Comment by cassytaylor** [comment]
Hey @Akarshit, this looks good to me and is design approved. 👍

**Comment by aaronjudd** [comment]
@jshimko love to turn around a quick review on this one...

**Comment by jshimko** [comment]
My open/pending review is above (although it appears to be out of date order).

In short, an `isLoading` prop should never be needed for a component wrapped with `composeWithTracker`.  One of the main features of that HOC is that it automatically shows a loading component until you run `onData` with your retrieved data props.

**Comment by zenweasel** [comment]
@jshimko Where is the pending review?

**Comment by aaronjudd** [comment]
@jshimko I think you need to submit the pending reviews so that others see your private comments..

**Comment by aaronjudd** [comment]
@Akarshit ETA on updates?

**Comment by Akarshit** [comment]
@aaronjudd sorry, I forgot to "submit" my comment. Will keep this updated from now on.

File: imports/plugins/core/checkout/client/containers/cartSubTotalContainer.js [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
-      cartSubTotal: cart.getSubTotal(),
-      cartCount: cart.getCount(),
-      cartShipping: cart.getShippingTotal(),
-      cartDiscount: cart.getDiscounts(),
-      cartTaxes: cart.getTaxTotal(),
-      cartTotal: cart.getTotal()
-    });
-  }
+  onData(null, {
+    isLoading: true
```
[the remark this answers is not in the document]
Comment (Akarshit) [reply imports/plugins/core/checkout/client/containers/cartSubTotalContainer.js]: @jshimko Is there any better way to forcefully display the loading spinner?

**Comment by zenweasel** [comment]
@jshimko I think we are waiting for follow-up from you

**Pull request merged**